certbot/certbot/configuration.py

```diff
@@ -78,17 +75,38 @@ def __init__(self, namespace: argparse.Namespace,
         # Check command line parameters sanity, and error out in case of problem.
         _check_config_sanity(self)
 
+    def set_argument_sources(self, argument_sources: Dict[str, ArgumentSource]) -> None:
+        """
+        Associate the NamespaceConfig with a dictionary describing where each of its arguments
+        came from. This is necessary for making runtime evaluations on whether an argument was
+        specified by the user or not (see `set_by_user`).
```

Comment:
Might be worth including an example of how to build the argument dict, by pointing how we do it. Something like "See `certbot._internal.cli.helpful.HelpfulArgumentParser._build_sources_dict` for an example of how to build this dictionary."